### PR TITLE
write document about fees in coinswap.

### PR DESCRIPTION
**Pull request by KnowWhoami**

### CODE REVIEW

File: v1/5_fees.md [hunk 1]
```diff
@@ -1,11 +1,85 @@
...
-2. **Amount Relative Fee** - This component is charged as a percentage of the total amount being swapped.
-   
-3. **Time Relative Fee** - This component is a multiple of the relative locktime period that the maker needs to wait in case the contract transaction is broadcasted to be able to claim the refund.
+## Types of Fees Taker Has to Pay
 
-Note that the this fee does not include the transaction fees and it is calculated separately.
\ No newline at end of file
+### 1) **Coinswap Fees**
+
+`Coinswap fees` are charged by Makers to Takers for facilitating the coinswap. These fees compensate Makers for their services and are calculated based on specific parameters. The formula for calculating the coinswap fee is:
```
Comment (rishkwal) [inline v1/5_fees.md:9]:
```suggestion
`Coinswap fees` are charged by Makers for facilitating the coinswap. These fees compensate Makers for their services and are calculated based on specific parameters. The formula for calculating the Coinswap fee is:
```

File: v1/5_fees.md [hunk 2]
```diff
@@ -1,11 +1,85 @@
...
+**Definitions:**
+- `swap_amount`: The amount the Taker wants to swap.
+- `REFUND_LOCKTIME`: The time duration the Maker must wait to claim a refund in case of an unsuccessful coinswap.
+
+**Fee Parameters for Coinswap:**
+- **BASE_FEE**: A fixed fee charged by the Maker for their service. This fee is constant and must be paid for any coinswap, regardless of other factors.
+
+```rust
+pub const BASE_FEE: u64 = 1000;
+```
```
Comment (rishkwal) [inline v1/5_fees.md:28]: Can be removed. You can just briefly mention that the default value is 1000

File: v1/5_fees.md [hunk 3]
```diff
@@ -1,11 +1,85 @@
...
+
+- **AMOUNT_RELATIVE_FEE_PCT**: A percentage fee based on the amount being swapped. As the amount increases, the fee becomes more significant, reflecting the increased value of the transaction and the risk involved for the Maker.
+
+```rust
+pub const AMOUNT_RELATIVE_FEE_PCT: f64 = 2.50;
+```
```
Comment (rishkwal) [inline v1/5_fees.md:34]: Again, just mention that the default value in a sentence

File: v1/5_fees.md [hunk 4]
```diff
@@ -1,11 +1,85 @@
...
+
+- **TIME_RELATIVE_FEE_PCT**: A percentage fee based on the `refund locktime`. The longer the `refund locktime`, the longer the Maker has to wait to claim a refund in case of an unsuccessful coinswap. Thus, this fee increases with the length of the locktime.
+
+```rust
+pub const TIME_RELATIVE_FEE_PCT: f64 = 0.10;
+```
```
Comment (rishkwal) [inline v1/5_fees.md:40]: Same as above

File: v1/5_fees.md [hunk 5]
```diff
@@ -1,11 +1,85 @@
...
+- **Amount Relative Fee** = (100,000 * 2.5) / 100 = 2,500 sats
+- **Time Relative Fee** = (100,000 * 20 * 0.1) / 100 = 2,000 sats
+
+**Total Fee:**
+
+```
+total_fee = 1,000 + 2,500 + 2,000 = 5,500 sats
+```
+
+> **Note**: These fee rates are designed to asymptotically approach `5-10%` of the swap amount as the swap amount and refund locktime increase.
```
Comment (rishkwal) [inline v1/5_fees.md:55]:
```suggestion

> **Note**: The default fee rates are designed to asymptotically approach `5-10%` of the swap amount as the swap amount and refund locktime increase.
```

File: v1/5_fees.md [hunk 6]
```diff
@@ -1,11 +1,85 @@
...
+
+---
+
+### 2) **Mining Fees**
+
+In addition to the Coinswap fees, the Taker is also responsible for paying the **Maker's mining fees**. These are the fees associated with the funding transactions that the Maker broadcasts as part of the coinswap. Each funding transaction requires a miner's fee to incentivize miners to include the transaction in a block. These fees are charged to the Taker and are typically paid separately from the Coinswap fees.
```
Comment (rishkwal) [inline v1/5_fees.md:61]:
```suggestion
In addition to the Coinswap fees, the Taker is also responsible for paying the **Maker's mining fees**. These are the fees associated with the funding transactions that the Maker broadcasts as part of the coinswap. Each funding transaction requires a miner's fee to incentivize miners to include the transaction in a block. These fees are charged from the Taker and paid separately from the Coinswap fees.
```

File: v1/5_fees.md [hunk 7]
```diff
@@ -1,11 +1,85 @@
...
+
+Once the Coinswap transaction is successfully completed, the Taker receives the swap amount minus all Coinswap fees and mining fees. This can be represented as:
+
+```text
+received_swap_amount = initial_swap_amount - all_coinswap_fees - all_mining_fees
+```
+
+Therefore, the Taker ultimately pays all these fees.
+
+However, if the Coinswap transaction fails, the scenario might differ. For more details on successful and unsuccessful Coinswap outcomes, refer to the integration tests available [here](https://github.com/citadel-tech/coinswap/tree/master/tests), which cover all possible outcomes of the Coinswap protocol.
```
Comment (rishkwal) [inline v1/5_fees.md:83]: The abort scenarios will be added to the security docs. Link them here.
```suggestion
However, if the Coinswap transaction fails, the scenario might differ. For more details on successful and unsuccessful Coinswap outcomes, refer to the abort scenarios [here](./6_security.md).
```

**Review by rishkwal — CHANGES_REQUESTED** [review]
ACK overall, the fees are explained well. Just a few NITs

**Comment by KnowWhoami** [comment]
@rishkwal 
Solved all  your reviews.

**Comment by rishkwal** [comment]
ACK

**Pull request merged**